packages/payment-experience-api/src/api/paytrailOnlineRefundPaymentSuccessController.ts

```diff
@@ -0,0 +1,170 @@
+
+    const refund = await getRefundAdmin({ refundId })
+    const orderId = refund.refund.orderId
+    const order = await getOrderAdmin({ orderId })
+    const merchantId = parseMerchantIdFromFirstOrderItem(order)
+
+    if (!merchantId) {
+      logger.error('Paytrail: No merchantId found from order')
+      throw new ExperienceError({
+        code: 'merchant-id-not-found',
+        message: 'No merchantId found from order.',
+        responseStatus: StatusCode.NotFound,
+        logLevel: 'info',
+      })
+    }
+
+    try {
+      const paytrailStatus = await checkPaytrailRefundCallbackUrl({
+        params: query,
+        merchantId: merchantId,
+      })
+      logger.debug(
+        `PaytrailStatus for refund ${refundId}: ${JSON.stringify(
+          paytrailStatus
+        )}`
+      )
+      if (!paytrailStatus.valid) {
+        logger.debug(
+          `PaytrailStatus is not valid for refund ${refundId}, redirect to failure url`
+        )
+        return result.redirect(
+          302,
+          PaytrailOnlineRefundPaymentSuccessController.getFailureRedirectUrl()
+        )
+      }
+
+      const redirectUrl = await createUserRedirectUrl({
+        order,
+        paymentReturnStatus: paytrailStatus,
+        redirectPaymentUrlBase: PaytrailOnlineRefundPaymentSuccessController.getRedirectUrl(),
+      })
+      // TODO: Make changes that sends email from refund
+      // Function contains internal checks when to send receipt.
+      // await sendReceiptToCustomer(paytrailStatus, orderId, order)
+
+      // Only cancel authorized card renewals.
+      if (isAuthorized(paytrailStatus) && isCardRenewal(paytrailStatus)) {
+        const cancelled = await this.cancelAuthorizationPayments(order)
+        // Error count more than 0
+        let cancelledCount = cancelled[1]
+        if (cancelledCount != 0) {
+          return result.redirect(
+            302,
+            PaytrailOnlineRefundPaymentSuccessController.getCardRenewalFailureRedirectUrl(
+              order.orderId
+            ).toString()
+          )
+        }
+      }
+
+      return result.redirect(302, redirectUrl.toString())
+    } catch (error) {
+      logger.error(error)
+      logger.debug(
+        `Error occurred, redirect user to failure url for order ${orderId}`
+      )
+      return result.redirect(
+        302,
+        PaytrailOnlineRefundPaymentSuccessController.getFailureRedirectUrl().toString()
+      )
+    }
+  }
+
+  private static getFailureRedirectUrl() {
+    const redirectUrl = this.checkAndCreateRedirectUrl()
+    redirectUrl.pathname = 'failure'
+
+    return redirectUrl.toString()
+  }
+
+  public static checkAndCreateRedirectUrl() {
+    if (!process.env.REDIRECT_PAYTRAIL_PAYMENT_URL_BASE) {
+      throw new Error('No default paytrail redirect url specified')
+    }
+    return new URL(process.env.REDIRECT_PAYTRAIL_PAYMENT_URL_BASE)
+  }
+
+  public static getRedirectUrl() {
+    if (!process.env.REDIRECT_PAYTRAIL_PAYMENT_URL_BASE) {
```

Comment:
We should add refund/ to paytrail payment url base to make it different form order failure urls

Reply:
Should it be with orderId like in **getCardRenewalFailureRedirectUrl** or without?